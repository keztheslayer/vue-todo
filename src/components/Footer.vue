<template>
    <footer class="todo__counters">
        <div class="todo__counter">
            Total: {{ totalItems }}
        </div>
        <button
            class="btn btn-primary btn-sm todo__clean-btn"
            @click="removeFinishedTasks"
        >
            Remove all finished tasks
        </button>
        <div class="todo__counter">
            Finished: {{ finishedTasks }}
        </div>
    </footer>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';

export default {
    props : {
        total : {
            type    : Number,
            default : 0,
        },
        finished : {
            type    : Number,
            default : 0,
        },
    },
    computed : {
        ...mapGetters( [
            'finishedTasks',
            'totalItems',
        ] ),
    },
    methods : {
        ...mapActions( [
            'removeFinishedTasks',
        ] ),
    },
};
</script>

<style lang="less">
.todo {
    &__counters {
      color: #333;
      font-size: 15px;
      display: flex;
      align-items: center;
      justify-content: space-between;

      @media (max-width: 768px){
        flex-direction: column;
      }
    }

    &__clean-btn {
      @media (max-width: 768px){
        order: -1;
        flex-grow: 1;
        margin-bottom: 12px;
      }
    }
}

</style>
