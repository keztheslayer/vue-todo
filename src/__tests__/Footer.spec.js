import Footer from '../components/Footer.vue';
import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

const localVue = createLocalVue();

localVue.use( Vuex );

describe( 'Footer.vue', () => {
    let getters;

    let store;

    let actions;

    beforeEach( () => {
        actions = {
            removeFinishedTasks : jest.fn(),
        };

        getters = {
            totalItems    : () => 3,
            finishedTasks : () => 1,
        };

        store = new Vuex.Store( {
            getters,
            actions,
        } );
    } );

    it( 'renders correct markup', () => {
        const wrapper = mount(
            Footer,
            {
                store,
                localVue,
            }
        );

        const total    = wrapper.find('.todo__counter_type_total');
        const finished = wrapper.find('.todo__counter_type_finished');

        expect( total.text() ).toEqual( expect.stringContaining( getters.totalItems().toString() ) );

        expect( finished.text() ).toEqual( expect.stringContaining( getters.finishedTasks().toString() ) );


        expect( wrapper.element ).toMatchSnapshot();
    } );

    it( 'dispatches "removeFinishedTasks" when clicking on clean-btn', () => {
        const wrapper = mount(
            Footer,
            {
                store,
                localVue,
            }, );
        const cleanButton = wrapper.find('.todo__clean-button');

        cleanButton.trigger('click');
        expect( actions.removeFinishedTasks ).toHaveBeenCalled();
    } );
} );
