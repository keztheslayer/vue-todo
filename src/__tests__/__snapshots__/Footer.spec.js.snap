// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue renders correct markup 1`] = `
<footer
  class="todo__counters"
>
  <div
    class="todo__counter todo__counter_type_total"
  >
    
        Total: 3
    
  </div>
   
  <button
    class="btn btn-primary btn-sm todo__clean-button"
  >
    
        Remove all finished tasks
    
  </button>
   
  <div
    class="todo__counter todo__counter_type_finished"
  >
    
        Finished: 1
    
  </div>
</footer>
`;
